### ISSUE TITLE
G4.0.2.1 - Saved session information doesn't get updated correctly on window close

### ISSUE THREAD

**Issue by CMDRNuffin** [MrAlex94/Waterfox#2445]
### What happened?

TL;DR: When quitting out of waterfox by closing the last window, recent navigation is not saved in the stored session.

This issue might be amplified by my rather large number of open tabs (27 at the time of writing).

I noticed that when I close the last remaining window shortly after navigating to a new page, the session that gets restored on the next start sometimes contains the old page instead. Some testing then revealed that this also applies to newly opened tabs (which are not saved at all).

Closed tabs are saved correctly, however. Which is annoying if I search something, open a bunch of promising results in new tabs for closer inspection, close the search tab and then decide to call it a day and resume later.

This does not happen

- on my installation of Firefox (Version 94.0.1)
- when I _Exit_ the program using the menu
- when I let the new page sit there for a while before exiting

### Reproducible?

- [X] I have tested this on Firefox.

### Version

G4.𝑋 (Future)

### What platform are you seeing the problem on?

Windows

**Comment by MrAlex94** [comment]
Do you still have this issue with G5?

**Comment by CMDRNuffin** [comment]
No, this seems to be working correctly now.
